scene/main/node.cpp

```diff
@@ -2260,22 +2255,8 @@ void Node::_duplicate_signals(const Node *p_original, Node *p_copy) const {
 	get_all_signal_connections(&conns);
 
 	for (List<Connection>::Element *E = conns.front(); E; E = E->next()) {
-
 		if (E->get().flags & CONNECT_PERSIST) {
-			//user connected
-			NodePath p = p_original->get_path_to(this);
-			Node *copy = p_copy->get_node(p);
-
-			Node *target = Object::cast_to<Node>(E->get().target);
-			if (!target) {
-				continue;
-			}
-			NodePath ptarget = p_original->get_path_to(target);
-			Node *copytarget = p_copy->get_node(ptarget);
-
-			if (copy && copytarget) {
-				copy->connect(E->get().signal, copytarget, E->get().method, E->get().binds, CONNECT_PERSIST);
-			}
+			p_copy->connect(E->get().signal, E->get().target, E->get().method, E->get().binds, E->get().flags);
```

Comment:
It was originally implemented by @reduz in 9b692b174b06ebb80d6e5a82309b9c9fc44c797d, not sure if there's a drawback to the simplified method you're using. @reduz, can you have a look?

Reply:
I better restore it to original, if its matter it could be subject of separate issue.

Reply:
I thought you meant those changes were needed for it to work?

Reply:
for fixing - #12750 this is not required

Reply:
This change affects only raw call of duplicate in script

Reply:
I guess if someone more experienced will touch this... I afraid my change could damage intended behavior of this function in some situations, its a node after all - a core of all godot classes